### PR TITLE
Immune deconv quantiseq

### PR DESCRIPTION
**Pull request by kelseykeith**
#### What scientific question is your analysis addressing?

Update the `immune-deconv` module to add quanTIseq and remove CIBERSORT as deconvolution options in order to to stay up to date with OpenPBTA <https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/immune-deconv>

#### What was your approach?

Made minimal changes to add `quanTIseq` as an option and to remove `CIBERSORT` as an option when running `immunedeconv`. Code is not completely identical to the OpenPBTA module since we run the deconvolution by cancer type.

#### What GitHub issue does your pull request address?

Updated analysis: immune-deconv to include quanTIseq #321 <https://github.com/PediatricOpenTargets/ticket-tracker/issues/321>

#### Which areas should receive a particularly close look?

1. Review the code
2. Test run the module with both the xcell and quanTIseq options. XCell results should be unchanged from previous since no changes were made there. 
3. Check both outputs for consistency.

#### Is there anything that you want to discuss further?

N/A

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

RDS files and PDF plots

#### What is your summary of the results?

- `results/quantiseq_output.rds`
- `results/xcell_output.rds`
- `plots/quantiseq_heatmap_by_group.pdf`
- `plots/xcell_heatmap_by_group.pdf`

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/immune-deconv/README.md [hunk 1]
```diff
@@ -30,15 +39,15 @@ data/histologies.tsv
...
 
 3. Output: 
 
 ```
 results/{deconv_method}_output.rds
-````
+```
 
-The results in the rds file are predicted immune scores per cell type per input sample. These scores are not actual cell fractions but arbitrary scores representing enrichment of the cell types which can be compared across various cancer/gtex groups. Depending on the user requirement, the output can be used to create various visualizations. 
+For `xCell`, the results in the rds file are predicted immune scores per cell type per input sample. These scores are not actual cell fractions but arbitrary scores representing enrichment of the cell types which can be compared across various cancer/gtex groups. The `quanTIseq` results, in contrast, provide an absolute score that can be interpreted as a cell fraction and the results in the rds file are the asbsolute scores per cell type per input sample. Depending on the user requirements, the output can also be used to create various visualizations.
```
Comment (aadamk) [inline analyses/immune-deconv/README.md]:
```suggestion
For `xCell`, the results in the rds file are predicted immune scores per cell type per input sample. These scores are not actual cell fractions but arbitrary scores representing enrichment of the cell types which can be compared across various cancer/gtex groups. The `quanTIseq` results, in contrast, provide an absolute score that can be interpreted as a cell fraction and the results in the rds file are the absolute scores per cell type per input sample. Depending on the user requirements, the output can also be used to create various visualizations. 
```

File: analyses/immune-deconv/01-immune-deconv.R [hunk 2]
```diff
@@ -1,5 +1,5 @@
-# Author: Komal S. Rathi
-# script to perform immune characterization using immunedeconv
+# Author: Komal S. Rathim updated, 2020-07 Kelsey Keith
```
Comment (aadamk) [inline analyses/immune-deconv/01-immune-deconv.R]:
```suggestion
# Author: Komal S. Rathi updated, 2020-07 Kelsey Keith
```

**Review by aadamk — APPROVED** [review]
thank you for this PR @kelseykeith 
i completed end-to-end testing and this module ran successfully for me. 
just a couple of typos to resolve before merging.

**Comment by jharenza** [comment]
Thanks for the review @aadamk , and work on this @kelseykeith! I would like to hold on the merge until we finalize GitHub actions testing data so this can also run through GA. I should be able to wrap this up in the next two days.

**Comment by jharenza** [comment]
@kelseykeith to move this one forward for our toolkit purposes, I merged dev and uncommented immune deconv in the YAML file. 🤞

**Comment by kelseykeith** [comment]
Great, I will try rerunning!

**Comment by jharenza** [comment]
You don't actually have to do anything else right now- I just wanted it to trigger a GitHub actions run. It failed, but not because of your module, because MRAN is down and the docker image could not build. We can restart the job and see how it goes, so we will wait on the GA for anything further!

**Comment by kelseykeith** [comment]
Ok, great, never mind then!

**Comment by jharenza** [comment]
@kelseykeith can you fix the merge conflicts with the analysis readme please?

**Comment by kelseykeith** [comment]
Done!

File: analyses/immune-deconv/run-immune-deconv.sh [hunk 3]
```diff
@@ -1,54 +1,47 @@
...
-# xCell
+### xCell
 # generate deconvolution output
-echo "Deconvolution..."
+echo "Deconvolution xCell..."
 Rscript --vanilla 01-immune-deconv.R \
---expr_mat '../../data/gene-expression-rsem-tpm-collapsed.rds' \
---clin_file '../../data/histologies.tsv' \
+--expr_mat '../../data/v11/gene-expression-rsem-tpm-collapsed.rds' \
+--clin_file '../../data/v11/histologies.tsv' \
```
Comment (jharenza) [inline analyses/immune-deconv/run-immune-deconv.sh]: please set these to use the data folder directly - I think this is why the module is failing

File: analyses/immune-deconv/run-immune-deconv.sh [hunk 4]
```diff
@@ -1,54 +1,47 @@
...
-	--output_dir 'plots'
-
-fi
\ No newline at end of file
+### quanTIseq
+# generate deconvolution output
+echo "Deconvolution quanTIseq..."
+Rscript --vanilla 01-immune-deconv.R \
+--expr_mat '../../data/v11/gene-expression-rsem-tpm-collapsed.rds' \
+--clin_file '../../data/v11/histologies.tsv' \
```
Comment (jharenza) [inline analyses/immune-deconv/run-immune-deconv.sh]: same comment - use the data dir

**Review by jharenza — CHANGES_REQUESTED** [review]
see comments about updating to use the data dir instead of a specific version

**Comment by kelseykeith** [comment]
Done! Sorry I forgot to change them back

**Review by jharenza — APPROVED** [review]
perfect, thank you!

**Pull request merged**